src/routes/InstancesPage/InstancesPage.js

```diff
@@ -130,11 +132,166 @@ function InstancesPage() {
     setFilters({});
   }
 
-  if (isFetching) {
-    return (
-      <Bullseye>
-        <Spinner />
-      </Bullseye>
+  let content = null;
+
+  if (
+    !isFetching &&
+    instances?.length === 0 &&
```

Comment:
Nit: I think all the cases of `instances?` can be replaced with just `instance` in this component too.